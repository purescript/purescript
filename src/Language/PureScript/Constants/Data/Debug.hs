module Language.PureScript.Constants.Data.Debug where

import Data.String (IsString)
import Language.PureScript.Names

pattern DataDebug :: ModuleName
pattern DataDebug = ModuleName "Data.Debug"

pattern Debug :: Qualified (ProperName 'ClassName)
pattern Debug = Qualified (ByModuleName DataDebug) (ProperName "Debug")

debug :: forall a. (IsString a) => a
debug = "debug"

identDebug :: Qualified Ident
identDebug = Qualified (ByModuleName DataDebug) (Ident debug)

pattern DataDebugType :: ModuleName
pattern DataDebugType = ModuleName "Data.Debug.Type"

int :: forall a. (IsString a) => a
int = "int"

identInt :: Qualified Ident
identInt = Qualified (ByModuleName DataDebugType) (Ident int)

number :: forall a. (IsString a) => a
number = "number"

identNumber :: Qualified Ident
identNumber = Qualified (ByModuleName DataDebugType) (Ident number)

boolean :: forall a. (IsString a) => a
boolean = "boolean"

identBoolean :: Qualified Ident
identBoolean = Qualified (ByModuleName DataDebugType) (Ident boolean)

char :: forall a. (IsString a) => a
char = "char"

identChar :: Qualified Ident
identChar = Qualified (ByModuleName DataDebugType) (Ident char)

string :: forall a. (IsString a) => a
string = "string"

identString :: Qualified Ident
identString = Qualified (ByModuleName DataDebugType) (Ident string)

array :: forall a. (IsString a) => a
array = "array"

identArray :: Qualified Ident
identArray = Qualified (ByModuleName DataDebugType) (Ident array)

record :: forall a. (IsString a) => a
record = "record"

identRecord :: Qualified Ident
identRecord = Qualified (ByModuleName DataDebugType) (Ident record)

constructor :: forall a. (IsString a) => a
constructor = "constructor"

identConstructor :: Qualified Ident
identConstructor = Qualified (ByModuleName DataDebugType) (Ident constructor)

opaque :: forall a. (IsString a) => a
opaque = "opaque"

identOpaque :: Qualified Ident
identOpaque = Qualified (ByModuleName DataDebugType) (Ident opaque)

opaque_ :: forall a. (IsString a) => a
opaque_ = "opaque_"

identOpaque_ :: Qualified Ident
identOpaque_ = Qualified (ByModuleName DataDebugType) (Ident opaque_)
