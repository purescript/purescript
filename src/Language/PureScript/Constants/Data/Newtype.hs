module Language.PureScript.Constants.Data.Newtype where

import Prelude.Compat
import Language.PureScript.Names

pattern Newtype :: Qualified (ProperName 'ClassName)
pattern Newtype = Qualified (Just (ModuleName "Data.Newtype")) (ProperName "Newtype")
