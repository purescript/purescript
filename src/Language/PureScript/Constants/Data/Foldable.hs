module Language.PureScript.Constants.Data.Foldable where

import Data.String (IsString)
import Language.PureScript.Names

foldl :: forall a. (IsString a) => a
foldl = "foldl"

foldr :: forall a. (IsString a) => a
foldr = "foldr"

foldMap :: forall a. (IsString a) => a
foldMap = "foldMap"

pattern DataFoldable :: ModuleName
pattern DataFoldable = ModuleName "Data.Foldable"

pattern Foldable :: Qualified (ProperName 'ClassName)
pattern Foldable = Qualified (ByModuleName DataFoldable) (ProperName "Foldable")

identFoldl :: Qualified Ident
identFoldl = Qualified (ByModuleName DataFoldable) (Ident foldl)

identFoldr :: Qualified Ident
identFoldr = Qualified (ByModuleName DataFoldable) (Ident foldr)

identFoldMap :: Qualified Ident
identFoldMap = Qualified (ByModuleName DataFoldable) (Ident foldMap)
