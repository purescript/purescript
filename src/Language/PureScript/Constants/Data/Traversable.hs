module Language.PureScript.Constants.Data.Traversable where

import Data.String (IsString)
import Language.PureScript.Names

traverse :: forall a. (IsString a) => a
traverse = "traverse"

sequence :: forall a. (IsString a) => a
sequence = "sequence"

pattern DataTraversable :: ModuleName
pattern DataTraversable = ModuleName "Data.Traversable"

pattern Traversable :: Qualified (ProperName 'ClassName)
pattern Traversable = Qualified (ByModuleName DataTraversable) (ProperName "Traversable")

identTraverse :: Qualified Ident
identTraverse = Qualified (ByModuleName DataTraversable) (Ident traverse)
