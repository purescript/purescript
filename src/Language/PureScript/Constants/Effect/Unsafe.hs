module Language.PureScript.Constants.Effect.Unsafe where

import Data.String (IsString)

effectUnsafe :: forall a. (IsString a) => a
effectUnsafe = "Effect_Unsafe"

unsafePerformEffect :: forall a. (IsString a) => a
unsafePerformEffect = "unsafePerformEffect"
