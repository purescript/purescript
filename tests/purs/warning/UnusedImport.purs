-- @shouldWarnWith UnusedImport
-- @shouldWarnWith UnusedImport
module Main where

import Data.Unit (Unit, unit)

-- All of the below are unused
import Effect
import Effect.Console as Console

main :: Unit
main = unit
