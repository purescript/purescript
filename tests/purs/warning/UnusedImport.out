Warning 1 of 2:

  in module [33mMain[0m
  at tests/purs/warning/UnusedImport.purs:8:1 - 8:14 (line 8, column 1 - line 8, column 14)

    The import of [33mEffect[0m is redundant


  See https://github.com/purescript/documentation/blob/master/errors/UnusedImport.md for more information,
  or to contribute content related to this warning.

Warning 2 of 2:

  in module [33mMain[0m
  at tests/purs/warning/UnusedImport.purs:9:1 - 9:33 (line 9, column 1 - line 9, column 33)

    The qualified import of [33mEffect.Console[0m as [33mConsole[0m is redundant


  See https://github.com/purescript/documentation/blob/master/errors/UnusedImport.md for more information,
  or to contribute content related to this warning.

